### ISSUE TITLE
can't store an or-ing of Bitflags in a const or static

### ISSUE THREAD

**Issue by sam0x17** [bitflags/bitflags#243]
```rust
const LEFT_SHARED_MASK: Buttons4 = Buttons4::MINUS | Buttons4::L_STICK | Buttons4::CAPTURE;
```
results in:
```
error[E0015]: calls in constants are limited to constant functions, tuple structs and tuple variants
   --> src\main.rs:151:36
    |
151 | const LEFT_SHARED_MASK: Buttons4 = Buttons4::MINUS | Buttons4::L_STICK | Buttons4::CAPTURE;
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by sam0x17** [comment]
btw I got around this using the lazy_static crate

**Comment by nickelc** [comment]
You can use `Flags::from_bits_truncate` and `Flags::A.bits` or `Flags::A.bits()` for this.

```rust
const MASK: Flags = Flags::from_bits_truncate(Flags::A.bits | Flags::B.bits | Flags::C.bits);
const MASK: Flags = Flags::from_bits_truncate(Flags::A.bits() | Flags::B.bits() | Flags::C.bits());
```

**Comment by KodrAus** [comment]
We've also got #244 currently open that would make this a little nicer to write until we can use `|` in const:

```rust
const LEFT_SHARED_MASK: Buttons4 = Buttons4::MINUS.union(Buttons4::L_STICK).union(Buttons4::CAPTURE);
```

**Comment by KodrAus** [comment]
We've just released `1.3.0` that supports the above syntax. The issue of being able to use real operators will be blocked upstream for a while, so I'll go ahead and close this one, but one day we should be able to support it!
